crates/hyperswitch_connectors/src/connectors/nexixpay.rs

```diff
@@ -465,17 +480,31 @@ impl ConnectorIntegration<Authorize, PaymentsAuthorizeData, PaymentsResponseData
         event_builder: Option<&mut ConnectorEvent>,
         res: Response,
     ) -> CustomResult<PaymentsAuthorizeRouterData, errors::ConnectorError> {
-        let response: nexixpay::NexixpayPaymentsResponse = res
-            .response
-            .parse_struct("NexixpayPaymentsResponse")
-            .change_context(errors::ConnectorError::ResponseDeserializationFailed)?;
-        event_builder.map(|i| i.set_response_body(&response));
-        router_env::logger::info!(connector_response=?response);
-        RouterData::try_from(ResponseRouterData {
-            response,
-            data: data.clone(),
-            http_code: res.status_code,
-        })
+        if data.request.connector_mandate_id().is_none() {
```

Comment:
this check won't be needed if we use an enum for the PaymentsResponse
`enum NexixpayResponse {
    PaymentsPresonse(NexixpayPaymentsResponse),
    MandateResponse(NexixpayMandateResponse),
}`
`

Reply:
Take reference from Fiuu: Transformers
```#[derive(Debug, Serialize, Deserialize)]
#[serde(untagged)]
pub enum FiuuPaymentsResponse {
    PaymentResponse(Box<PaymentsResponse>),
    QRPaymentResponse(Box<DuitNowQrCodeResponse>),
    Error(FiuuErrorResponse),
    RecurringResponse(Vec<Box<FiuuRecurringResponse>>),
}
```